services/web/server/src/simcore_service_webserver/notifications/_db_comp_tasks_listening_task.py

```diff
@@ -54,7 +53,71 @@ async def _update_project_state(
     await projects_api.notify_project_state_update(app, project)
 
 
-async def listen(app: web.Application, db_engine: Engine):
+@dataclass(frozen=True)
+class _CompTaskNotificationPayload:
+    data: dict
+    changes: dict
+
+
+async def _handle_db_notification(
+    app: web.Application, payload: _CompTaskNotificationPayload, conn: SAConnection
+) -> None:
+    task_data = payload.data
+    task_changes = payload.changes
+
+    project_uuid = task_data.get("project_id", "undefined")
+    node_uuid = task_data.get("node_id", "undefined")
+
+    try:
+        # NOTE: we need someone with the rights to modify that project. the owner is one.
+        # find the user(s) linked to that project
+        the_project_owner = await _get_project_owner(conn, project_uuid)
+
+        if any(f in task_changes for f in ["outputs", "run_hash"]):
```

Comment:
MINOR: I think this  alternative reads very well and you avoid one extra check :-)
```python
if (new_outputs:= task_data.get("outputs", {})) or (new_run_hash:=task_data.get("run_hash", None) ):
     ...
     
```

Reply:
I did not rework the code. only refactor. I think I will keep it as it is for now.